waddrmgr/address.go

```diff
@@ -633,9 +652,12 @@ func (a *scriptAddress) Script() ([]byte, error) {
 	return a.unlock(a.manager.rootManager.cryptoKeyScript)
 }
 
+// Enforce scriptAddress satisfies the ManagedScriptAddress interface.
+var _ ManagedScriptAddress = (*scriptAddress)(nil)
+
 // newScriptAddress initializes and returns a new pay-to-script-hash address.
 func newScriptAddress(m *ScopedKeyManager, account uint32, scriptHash,
-	scriptEncrypted []byte) (*scriptAddress, error) {
+	scriptEncrypted []byte) (ManagedScriptAddress, error) {
```

Comment:
nit: Usually it's better to follow the rule of `accept interfaces, return structs` so the caller knows exactly what s/he is getting without the need of try/error casting.